### PR TITLE
AWS Cloud atomics

### PR DESCRIPTION
**Pull request by patel-bhavin**
**Details:**
Threat Research team at Splunk is actively working on creating atomic tests for AWS, this PR includes:

- Create a New AWS IAM user (T1136.003)
- Create AWS Group and add a user to that group. (T1098)
- Create AWS Access Key and Secret Key (T1098.001)

**Note:**
- The atomics use AWS CLI commands to execute simulations
- Updated supported platforms/executor name as per the new spec

**Testing:**
-  To test this, configure AWS CLI and have a `default` profile stanza
-   Test commands: 

`Invoke-AtomicTest -PathToAtomicsFolder <path> T1136.003`
`Invoke-AtomicTest -PathToAtomicsFolder <path> T1136.003 -Cleanup `

We are looking forward and stoked get feedback on how we can have atomic tests for Cloud Environments

### CODE REVIEW

File: atomics/T1098.001/T1098.001.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+display_name: Account Manipulation - Additional Cloud Credentials
+atomic_tests:
+- name: Create AWS Access Key and Secret Key
+  auto_generated_guid: 8822c3b0-d9f9-4daf-a043-491160a31122
+  description: |
+    Adversaries create their own new access and secret keys to programatically interact with AWS environment
+  supported_platforms:
+  - macos
+  - linux
+  cloud_provider:
```
Comment (patel-bhavin) [inline atomics/T1098.001/T1098.001.yaml:11]: remove this new field `cloud_provider`, instead add it to `supported_platforms` and update the code base to support new options

File: atomics/T1098/T1098.yaml [hunk 2]
```diff
@@ -101,3 +101,34 @@ atomic_tests:
...
+        echo Please run atomic test T1136.003, before running this atomic
+  executor:
+    command: |
+      aws iam create-group --group-name #{victim}
+      aws iam add-user-to-group --user-name #{victim} --group-name #{victim}
+    cleanup_command: |
+      aws iam remove-user-from-group --user-name #{victim} --group-name #{victim}
+      aws iam delete-group --group-name #{victim}
+
+    name: sh
```
Comment (patel-bhavin) [inline atomics/T1098/T1098.yaml:133]: Add new options for AWS and AZURE CLI parameters

File: atomics/T1098.001/T1098.001.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  executor:
+    command: |
+      aws iam create-access-key --user-name #{victim} > $PathToAtomicsFolder/T1098.001/bin/aws_secret.creds
+      cd $PathToAtomicsFolder/T1098.001/bin/
+      ./aws_secret.sh
+    cleanup_command: |
+      access_key=`cat $PathToAtomicsFolder/T1098.001/bin/aws_secret.creds| jq -r '.AccessKey.AccessKeyId'`
+      aws iam delete-access-key --access-key-id $access_key --user-name #{victim}
+      rm $PathToAtomicsFolder/T1098.001/bin/aws_secret.creds
+    name: aws
```
Comment (amashinchi-rc) [inline atomics/T1098.001/T1098.001.yaml:31]: I think the `name` here is `sh`? (as these are not using only the aws-cli executor.

File: atomics/T1098/T1098.yaml [hunk 4]
```diff
@@ -101,3 +101,30 @@ atomic_tests:
...
+        aws iam list-users | grep #{victim}
+      get_prereq_command: |
+        echo Please run atomic test T1136.003, before running this atomic
+  executor:
+    command: |
+      aws iam create-group --group-name #{victim}
+      aws iam add-user-to-group --user-name #{victim} --group-name #{victim}
+    cleanup_command: |
+      aws iam remove-user-from-group --user-name #{victim} --group-name #{victim}
+      aws iam delete-group --group-name #{victim}
```
Comment (amashinchi-rc) [inline atomics/T1098/T1098.yaml:128]: I think you can remove the leading `aws` in these commands as the `aws` executor is declare in the `name`?

(example of powershell: https://github.com/redcanaryco/atomic-red-team/blob/058b5c2423c4a6e9e226f4e5ffa1a6fd9bb1a90e/atomics/T1547.004/T1547.004.yaml#L57 )

**Review by MHaggis — APPROVED** [review]

**Pull request merged**